cmd/minishift/cmd/start_preflight.go

```diff
@@ -0,0 +1,304 @@
+		fmt.Println(errorMessage)
+	} else {
+		atexit.ExitWithMessage(1, errorMessage)
+	}
+}
+
+func preflightCheckSucceedsOrFailsWithDriver(configNameOverrideIfSkipped string, execute preflightCheckWithDriverFunc, driver drivers.Driver, message string, treatAsWarning bool, configNameOverrideIfWarning string, errorMessage string) {
+	fmt.Printf("-- %s ... ", message)
+
+	isConfiguredToSkip := viper.GetBool(configNameOverrideIfSkipped)
+	isConfiguredToWarn := viper.GetBool(configNameOverrideIfWarning)
+
+	if isConfiguredToSkip {
+		fmt.Println("SKIP")
+		return
+	}
+
+	if execute(driver) {
+		fmt.Println("OK")
+		return
+	}
+
+	fmt.Println("FAIL")
+	errorMessage = fmt.Sprintf("   %s", errorMessage)
+	if isConfiguredToWarn || treatAsWarning {
+		fmt.Println(errorMessage)
+	} else {
+		atexit.ExitWithMessage(1, errorMessage)
+	}
+}
+
+func checkXhyveDriver() bool {
+	path, err := exec.LookPath("docker-machine-driver-xhyve")
+
+	if err != nil {
+		return false
+	}
+
+	fi, _ := os.Stat(path)
+	// follow symlinks
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false
+		}
+	}
+	fmt.Println("\n   Driver is available at", path)
+
+	fmt.Printf("   Checking for setuid bit ... ")
+	if fi.Mode()&os.ModeSetuid == 0 {
+		return false
+	}
+
+	return true
+}
+
+func checkKvmDriver() bool {
+	path, err := exec.LookPath("docker-machine-driver-kvm")
+	if err != nil {
+		return false
+	}
+	fmt.Printf(fmt.Sprintf("\n   Driver is available at %s ... ", path))
+
+	return true
+}
+
+func checkHypervDriver() bool {
+	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
+	if switchEnv == "" {
+		return false
+	}
+	return true
+}
+
+// Check to make sure the instance has an IPv4 address.
+// HyperV will issue IPv6 addresses on Internal virtual switch
+// https://github.com/minishift/minishift/issues/418
+func checkInstanceIP(driver drivers.Driver) bool {
+	ip, err := driver.GetIP()
+	if err == nil && net.ParseIP(ip).To4() != nil {
+		return true
+	}
+	return false
+}
+
+//
+func checkVMConnectivity(driver drivers.Driver) bool {
+	// used to check if the host can reach the VM
+	ip, _ := driver.GetIP()
+
+	cmd := exec.Command("ping", "-n 1", ip)
+	stdoutStderr, err := cmd.CombinedOutput()
+	if err != nil {
+		return false
+	}
+	fmt.Printf("%s\n", stdoutStderr)
+	return false
+}
+
+// Checks if the VM has connectivity to the outside network
+func checkIPConnectivity(driver drivers.Driver) bool {
+	ipToPing := viper.GetString(startFlags.CheckNetworkPingHost.Name)
+	if ipToPing == "" {
+		ipToPing = "8.8.8.8"
+	}
+
+	fmt.Printf("\n   Pinging %s ... ", ipToPing)
+	return miniutil.IsIPReachable(driver, ipToPing, false)
+}
+
+// Allows to test outside connectivity and possible proxy support
+func checkHttpConnectivity(driver drivers.Driver) bool {
+	urlToRetrieve := viper.GetString(startFlags.CheckNetworkHttpHost.Name)
+	if urlToRetrieve == "" {
+		urlToRetrieve = "http://minishift.io/index.html"
```

Comment:
I am still concerned about URL `http://minishift.io/index.html`. But it is ok as this test if failed will show a warning.